### PR TITLE
PR E2E test

### PR DESCRIPTION
**Pull request by psturc**
### JIRA
https://issues.redhat.com/browse/PLNSRVCE-149

### What
* updated PR pipeline to build & push bundles to `pull-request-builds` repo
* after bundles are pushed, configmap `build-pipelines-defaults` is updated to point to the pipeline bundle that was built
* E2E tests are triggered - pipeline used in E2E test should be extracted from the pipeline bundle ^

### Blockers
- [x] **[this PR](https://github.com/redhat-appstudio/build-service/pull/4) that will allow E2E tests to use the pipeline bundle defined in the configmap from the same namespace where apps and components are created.**
- [x] [this PR](https://github.com/redhat-appstudio/e2e-tests/pull/47) - allows to run E2E tests in a specific namespace with a service account that has just basic (namespace-scoped) permissions set
- [x] creation of a dockerconfig-type secret `redhat-appstudio-registry-pull-secret` in the same namespace where these tests will run (currently we're using `build-templates` ns)
- [x] creation of a secret containing a value for `GITHUB_TOKEN` env var used in E2E test

### CODE REVIEW

File: hack/build-and-push.sh
Comment (srivickynesh) [inline hack/build-and-push.sh:73]: line 51. Maybe move this below 26

File: .tekton/pull-request.yaml
Comment (psturc) [inline .tekton/pull-request.yaml:88]: https://github.com/redhat-appstudio/e2e-tests/pull/47

Comment (psturc) [reply .tekton/pull-request.yaml:88]: TODO: update this to `quay.io/redhat-appstudio/e2e-tests:latest` once the PR ^ gets merged

File: .tekton/pull-request.yaml
Comment (psturc) [inline .tekton/pull-request.yaml:97]: once we have a secret with github token, we could update this to something like:
```suggestion
                valueFrom:
                  secretKeyRef:
                    name: secret-name
                    key: bot-token
```

File: .tekton/pull-request.yaml
Comment (psturc) [inline .tekton/pull-request.yaml:102]: probably needs to be updated to
```suggestion
              - name: E2E_APPLICATIONS_NAMESPACE
```
but I'll update it once https://github.com/redhat-appstudio/e2e-tests/pull/47 is merged

File: hack/util-install-bundle.sh
Comment (Michkov) [inline hack/util-install-bundle.sh:5]: default should be current namespace, not the 'build-templates'

File: hack/build-and-push.sh
Comment (Michkov) [inline hack/build-and-push.sh:83]: you can remove this line, not sure what it does, but basically when `INSTALL_BUNDLE_NS` is not passed to the script then it's empty and util-install-bundle.sh will be using default.

Comment (psturc) [reply hack/build-and-push.sh:83]: It's a shorthand for `INSTALL_BUNDLE_NS=${INSTALL_BUNDLE_NS:-build-templates}`. But it makes sense to remove it considering the comments in [util-install-bundle](https://github.com/redhat-appstudio/build-definitions/pull/57/files/a31d98d16dda3b9b2d25ffedad32caf0c37790b9#diff-0cf40ce7cea04e5856c097541c8f9bd5b99c0e888ef7e78e43e37304ca9954b2) script

File: hack/util-install-bundle.sh
Comment (psturc) [reply hack/util-install-bundle.sh]: 👍  fixed

**Review by Michkov — APPROVED** [review]

**Comment by psturc** [comment]
After another discussion we decided to put this PR on hold (the reason is mentioned in the description)

**Comment by psturc** [comment]
E2E test keeps failing, because the build-service controller gets restarted after a component with git source is created:

```
controllers.ComponentInitialBuild	Waiting for devfile model in component: build-templates-e2e/build-suite-test-component-git-source	{"ComponentInitialBuild": "build-templates-e2e/build-suite-test-component-git-source"}
```
Full log [here](https://gist.github.com/psturc/24c8b634503f72e41b1536fab3dd4d65)

Hi @mmorhun any idea what could cause this issue? Current build-service image tag deployed on a staging cluster is `b59d88ee855dc99f2b2d8a45d20df5d1cf1663d9`

**Comment by psturc** [comment]
@Michkov tests updated to point to `redhat-appstudio/test-images` repo and they are now passing.

I had to update [build-and-push.sh](https://github.com/redhat-appstudio/build-definitions/pull/57/files#diff-78a47706e6b9aca3579f93bc34adac628f9c102de7257a2fe14939130786a807) script after rebase, so it might be worth to re-review it. Otherwise I think this PR is ready to merge.

**Review by Michkov — APPROVED** [review]

**Pull request merged**
